### PR TITLE
rv-ified VEP and Nirvana

### PR DESCRIPTION
**Pull request by cseed**
Builds on: https://github.com/hail-is/hail/pull/2711 (genericintervals7)

You probably want to wait until that goes in to review.

@konradjk Unfortunately, we don't have automated tests for VEP yet.  I'm bump the priority on that to make changing VEP safer, but in the mean time, can I ask you to run this on a small example to make sure we didn't break anything?  Thanks!

@jbloom22 Can you do the same for Nirvana?

### CODE REVIEW

**Comment by konradjk** [comment]
Fails with:
```
Java stack trace:
org.apache.spark.SparkException: Job aborted due to stage failure: Task 491 in stage 1.0 failed 20 times, most recent failure: Lost task 491.19 in stage 1.0 (TID 17951, exomes-sw-gf9k.c.broad-mpg-gnomad.internal): java.lang.IndexOutOfBoundsException: 3
	at is.hail.annotations.UnsafeRow.isNullAt(UnsafeRow.scala:294)
	at is.hail.annotations.UnsafeRow.get(UnsafeRow.scala:254)
	at is.hail.variant.MatrixTable$$anonfun$59$$anonfun$apply$42.apply(MatrixTable.scala:871)
	at is.hail.variant.MatrixTable$$anonfun$59$$anonfun$apply$42.apply(MatrixTable.scala:860)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$9$$anon$5.next(OrderedRVD.scala:637)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$9$$anon$5.next(OrderedRVD.scala:631)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$9$$anon$5.foreach(OrderedRVD.scala:631)
	at is.hail.io.RichRDDRegionValue$.writeRowsPartition(RowStore.scala:510)
	at is.hail.io.RichRDDRegionValue$$anonfun$writeRows$extension$1.apply(RowStore.scala:526)
	at is.hail.io.RichRDDRegionValue$$anonfun$writeRows$extension$1.apply(RowStore.scala:526)
	at is.hail.utils.richUtils.RichRDD$$anonfun$5.apply(RichRDD.scala:207)
	at is.hail.utils.richUtils.RichRDD$$anonfun$5.apply(RichRDD.scala:198)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsWithIndex$1$$anonfun$apply$25.apply(RDD.scala:820)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsWithIndex$1$$anonfun$apply$25.apply(RDD.scala:820)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:319)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:283)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:70)
	at org.apache.spark.scheduler.Task.run(Task.scala:86)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:274)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)

Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1454)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1442)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1441)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1441)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:811)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:811)
	at scala.Option.foreach(Option.scala:257)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:811)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:1667)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1622)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1611)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:632)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:1873)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:1886)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:1899)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:1913)
	at org.apache.spark.rdd.RDD$$anonfun$collect$1.apply(RDD.scala:912)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:358)
	at org.apache.spark.rdd.RDD.collect(RDD.scala:911)
	at is.hail.utils.richUtils.RichRDD$.writePartitions$extension(RichRDD.scala:209)
	at is.hail.io.RichRDDRegionValue$.writeRows$extension(RowStore.scala:526)
	at is.hail.variant.MatrixTable.write(MatrixTable.scala:2393)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:237)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:280)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:214)
	at java.lang.Thread.run(Thread.java:748)java.lang.IndexOutOfBoundsException: 3
	at is.hail.annotations.UnsafeRow.isNullAt(UnsafeRow.scala:294)
	at is.hail.annotations.UnsafeRow.get(UnsafeRow.scala:254)
	at is.hail.variant.MatrixTable$$anonfun$59$$anonfun$apply$42.apply(MatrixTable.scala:871)
	at is.hail.variant.MatrixTable$$anonfun$59$$anonfun$apply$42.apply(MatrixTable.scala:860)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$9$$anon$5.next(OrderedRVD.scala:637)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$9$$anon$5.next(OrderedRVD.scala:631)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at is.hail.rvd.OrderedRVD$$anonfun$apply$9$$anon$5.foreach(OrderedRVD.scala:631)
	at is.hail.io.RichRDDRegionValue$.writeRowsPartition(RowStore.scala:510)
	at is.hail.io.RichRDDRegionValue$$anonfun$writeRows$extension$1.apply(RowStore.scala:526)
	at is.hail.io.RichRDDRegionValue$$anonfun$writeRows$extension$1.apply(RowStore.scala:526)
	at is.hail.utils.richUtils.RichRDD$$anonfun$5.apply(RichRDD.scala:207)
	at is.hail.utils.richUtils.RichRDD$$anonfun$5.apply(RichRDD.scala:198)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsWithIndex$1$$anonfun$apply$25.apply(RDD.scala:820)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitionsWithIndex$1$$anonfun$apply$25.apply(RDD.scala:820)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:319)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:283)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:70)
	at org.apache.spark.scheduler.Task.run(Task.scala:86)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:274)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)


Hail version: devel-df17cef
Error summary: IndexOutOfBoundsException: 3
```
(NB: a custom VEP/LOFTEE, but that shouldn't matter - ran same thing on `devel-cd48e11` and it worked fine)

**Comment by cseed** [comment]
@konradjk OK, should be fixed.  Can you try again?

**Comment by jbloom22** [comment]
Testing on Nirvana now

**Comment by jbloom22** [comment]
Successfully ran pipeline in #2377 and found identical output.

File: src/main/scala/is/hail/methods/Nirvana.scala [hunk 1]
```diff
@@ -327,18 +328,44 @@ object Nirvana {
...
+      TStruct(
+        "pk" -> vds.locusType,
+        "v" -> vds.vSignature,
+        "vep" -> nirvanaSignature))
+
+    val nirvanaRowType = nirvanaOrderedRVType.rowType
+    val (t, pkProjection) = Type.partitionKeyProjection(vds.vSignature)
+    assert(t == vds.locusType)
+
+    val vepRVD: OrderedRVD = OrderedRVD(
```
Comment (jbloom22) [inline src/main/scala/is/hail/methods/Nirvana.scala]: nirvanaRVD

File: src/main/scala/is/hail/methods/Nirvana.scala [hunk 2]
```diff
@@ -327,18 +328,44 @@ object Nirvana {
...
+      nirvanaOrderedRVType,
+      vds.rdd2.partitioner,
+      annotations.mapPartitions { it =>
+        val region = Region()
+        val rvb = new RegionValueBuilder(region)
+        val rv = RegionValue(region)
+
+        it.map { case (v, vep) =>
```
Comment (jbloom22) [inline src/main/scala/is/hail/methods/Nirvana.scala]: (v, nirvana)

**Pull request merged**